File: MoviesAPIClient/TMDB/Models/TV/SeasonInfo.swift
//
//  SeasonInfo.swift
//  MoviesAPIClient
//
//  Created by Aryan Rogye on 9/14/26.
//

struct SeasonInfo: Decodable {
    let id: Int
    let airDate: String?
    let name: String
    let overview: String
    let posterPath: String?
    let seasonNumber: Int
    let episodes: [Episode]
}

struct Episode: Decodable, Identifiable, Equatable {
    let id: Int
    let name: String
    let overview: String

    let episodeNumber: Int
    let seasonNumber: Int

    let airDate: String?
    let runtime: Int?
    let stillPath: String?
}

File: MoviesShared/sharedLogic/src/commonMain/kotlin/com/aryanrogye/movies_shared/models/SeasonInfo.kt
package com.aryanrogye.movies_shared.models

import kotlinx.serialization.Serializable

@Serializable
data class KTSeasonInfo(
    val id: Int,
    val airDate: String? = null,
    val name: String,
    val overview: String,
    val posterPath: String? = null,
    val seasonNumber: Int,
    val episodes: List<KTEpisode>
)

@Serializable
data class KTEpisode(
    val id: Int,
    val name: String,
    val overview: String,
    val episodeNumber: Int,
    val seasonNumber: Int,
    val airDate: String? = null,
    val runtime: Int? = null,
    val stillPath: String? = null
)